### ISSUE TITLE
ZMap abruptly ends multi-port scans too early

### ISSUE THREAD

**Issue by phillip-stephens** [zmap/zmap#904]
**Describe the bug**
I want to scan multiple ports across the entire `IPv4` address space. ZMap ends without fully scanning, it looks like it's an issue with using `uint32` instead of `uint64`.

**CLI Arguments**

```
sudo ./zmap-dev/zmap/src/zmap -p 80-84 --dryrun -X -B 100G  -T 1 --metadata-file=/Users/phillip/Desktop/meta 2>&1 | tee ~/Desktop/logs
```

**Observed Behavior**
```
 4:51 23% (16m left); send: 4267050958 14.74 Mp/s (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:52 23% (16m left); send: 4281874854 14.75 Mp/s (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:53 97% (8s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:54 98% (7s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:55 98% (6s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:56 98% (5s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:57 99% (4s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:58 99% (3s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 4:59 99% (2s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 5:00 100% (1s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
 5:01 100% (0s left); send: 4294967295 done (14.65 Mp/s avg); recv: 0 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.00%
```
Notice how soon as we hit `4294967295`, scan progress stops

**Expected behavior**
Scan should complete over full IP x Port space

**Comment by phillip-stephens** [comment]
First seen by @annalittle, I'll take a look at this. Hoping we can just increase our `uint32_t` to `uint64_t` and it'll be a clean fix.

**Comment by phillip-stephens** [comment]
Closed by #905

**Comment by Outcalled** [comment]
This is still happening even when building the latest src, running on a dedicated server on Ubuntu 22.04

Command used: ./zmap -p12501,15001,17501,22501,25001,27501,32501,35001,37501,42501,45001,47501,50001 -B 3G --sender-threads=5 --dedup-window-size=9999999 --output-filter="success = 1 && repeat = 0" --output-fields="saddr,sport" -o "moreisps2.txt"

It checks 9% and then stops checking, I hope this gets fixed at some point.
